### DISCUSSION TITLE
Outside connection (tech)

### DISCUSSION THREAD

**Discussion by Pyth3rEx** [EveronLife/EveronLife#202] [Ideas]
Find a way to transfer data in and out of the game.

This would allow to display stats on a website (like the ALiVE waroom system on A3), have admins do work from an external console without having to connect on the server or allow some ease in political gameplay (eg. a mayor passing laws).

**Comment by Arkensor** [comment]
This is going to be covered by the web api that is also used as persistence data sink for SQL and NoSQL dbs. Information can go both ways through the built-in web api. Long polling to fetch events and aggregated calls for statistics and logging that the server sends to his known api endpoint. The technical means will be the easy part. The question would be what kind of data we want to collect and what kind of actions you want to execute from the web on the server. If we can compile a list for each then we can cover a lot of them with the base framework already.

**Reply by Pyth3rEx** [reply]
I think it would be best to have a good api than have lots of data transfer by default in the framework. This said a few examples can be good to implement for devs to build atop of.

Some good ideas such as #83 , #125 or #135 could make use of this system with an online 'interpol' style page or leaderboard stats pages.

**Reply by Pyth3rEx** [reply]
As I was thinking, even other things such as #37 could make use of this system by having an online licenseplates system or even having an online shop
